testing/theory/bdd.md

```diff
@@ -0,0 +1,29 @@
+# Behaviour Driven Development
+
+Behaviour Driven Development is a form of test driven development that uses software api's organized as plain language to write tests that can be read like sentences.
+
+Let's take our example below. Read over the code, and see if you can understand what is going on:
```

Comment:
I actually think Behat would be a better choice here, I'm not super familiar with PHPSpec but I'll caveat until a behat section can be written

```suggestion
Behaviour Driven Development is a form of test driven development that uses software api's organized as plain language to write tests that can be read like sentences.

There are two popular approaches to BDD with PHP in WordPress:

 * Behat, which uses the gherkin language to write tests that non-programmers can read
 * PHPSpec, which uses PHP to define tests
 
 Here we will use PHPSpec as an example
 
### PHPSpec 

Let's take our example below. Read over the code, and see if you can understand what is going on:
```